src/v/kafka/server/usage_manager.cc

```diff
@@ -0,0 +1,384 @@
+        /// window interval
+        const auto begin = std::chrono::seconds(buckets[*open_index].begin);
+        const auto now_ts = ss::lowres_system_clock::now();
+        const auto now = std::chrono::seconds(epoch_time_secs(now_ts));
+        const auto delta = now - begin;
+        if (begin > now || delta < _usage_window_width_interval) {
+            /// Clock drift has resulted in the later time appearing before the
+            /// beginning timestamp or time delta is within interval, continue
+            /// from open bucket
+            _current_window = *open_index;
+            last_window_delta = delta;
+        } else {
+            /// Close window and open a new one
+            buckets[*open_index].end = epoch_time_secs(now_ts);
+            _current_window = (*open_index + 1) % buckets.size();
+            buckets[_current_window].reset(now_ts);
+        }
+    }
+    _buckets = std::move(buckets);
+    return last_window_delta;
+}
+
+usage_manager::usage_manager(ss::sharded<storage::api>& storage)
+  : _usage_enabled(config::shard_local_cfg().enable_usage.bind())
+  , _usage_num_windows(config::shard_local_cfg().usage_num_windows.bind())
+  , _usage_window_width_interval(
+      config::shard_local_cfg().usage_window_width_interval_sec.bind())
+  , _usage_disk_persistance_interval(
+      config::shard_local_cfg().usage_disk_persistance_interval_sec.bind())
+  , _storage(storage) {
+    _usage_enabled.watch([this] { (void)reset(); });
+    _usage_num_windows.watch([this] { (void)reset(); });
+    _usage_window_width_interval.watch([this] { (void)reset(); });
+    _usage_disk_persistance_interval.watch([this] { (void)reset(); });
+}
+
+ss::future<> usage_manager::reset() {
+    oncore_debug_verify(_verify_shard);
+    try {
+        auto h = _background_gate.hold();
+        auto u = _background_mutex.get_units();
+        if (_accounting_fiber) {
+            /// Deallocate the accounting_fiber if the feature is disabled,
+            /// otherwise it will keep in memory the number of configured
+            /// historical buckets
+            auto accounting_fiber = std::exchange(_accounting_fiber, nullptr);
+            co_await accounting_fiber->stop();
+        }
+        co_await start();
+    } catch (ss::gate_closed_exception&) {
+        // shutting down
+    }
+}
+
+ss::future<> usage_manager::start() {
+    if (ss::this_shard_id() != usage_manager_main_shard) {
+        co_return; /// Async work only occurs on shard-0
+    }
+    if (_accounting_fiber) {
+        co_return; /// Double start called, do-nothing
+    }
+    if (!_usage_enabled()) {
+        co_return; /// Feature is disabled in config
+    }
+    _accounting_fiber = std::make_unique<accounting_fiber>(
+      this->container(),
+      _storage,
+      _usage_num_windows(),
+      _usage_window_width_interval(),
+      _usage_disk_persistance_interval());
+
+    co_await _accounting_fiber->start();
```

Comment:
it looks like this start can race with accounting fiber's stop() arriving asynchronous from a configuration binding change. am i missing something?